app/cli/cli.ts

```diff
@@ -1,57 +1,50 @@
 import readline from 'readline';
 import chalk from 'chalk';
-import { ClientManager } from '../src/client/manager.js';
-import { logger } from '../src/utils/logger.js';
-import { ILLMService } from '../src/ai/llm/services/types.js';
-import { AgentConfig } from '../src/config/types.js';
-import { initializeServices } from '../src/utils/service-initializer.js';
-import { AgentEventManager } from '../src/ai/llm/events/event-manager.js';
-import { CLISubscriber } from './cli-subscriber.js';
+import { ClientManager } from '../../src/client/manager.js'; // Adjusted path
+import { logger } from '../../src/utils/logger.js'; // Adjusted path
+import { ILLMService } from '../../src/ai/llm/services/types.js'; // Adjusted path
+import { AgentEventManager } from '../../src/ai/llm/events/event-manager.js'; // Adjusted path
+import { CLISubscriber } from './cli-subscriber.js'; // Now points to the new location
 
 /**
- * Start AI-powered CLI with unified configuration
- * @param config Agent configuration including MCP servers and LLM settings
- * @param connectionMode Whether to enforce all connections must succeed ("strict") or allow partial success ("lenient")
+ * Start the AI CLI using initialized services.
+ * This function is now called by the main entry point.
  */
 export async function initializeAiCli(
-    config: AgentConfig,
-    connectionMode: 'strict' | 'lenient' = 'lenient'
+    clientManager: ClientManager,
+    llmService: ILLMService
 ) {
-    try {
-        // Initialize services using the utility function
-        const { clientManager, llmService } = await initializeServices(config, connectionMode);
-        
-        // Run AI CLI
-        await runAiCli(clientManager, llmService);
-    } catch (error) {
-        logger.error(`Error running AI CLI: ${error.message}`);
-        process.exit(1);
-    }
+    // The main initialization is done in the shared initializeAgent function.
+    // This function now assumes services are already initialized.
+    await runAiCli(clientManager, llmService);
 }
```

Comment:
if `initializeAiCli` is just directly calling `runAICli`, we can delete `initializeAiCli` and directly use `runAiCli` instead